src/builder.rs

```diff
@@ -254,305 +254,305 @@ impl Builder {
     // TODO: Possibly make this generic over sign via struct metadata or subtypes
     // SubType: <I: IntSubType>(&self, lhs: &IntValue<I>, rhs: &IntValue<I>, name: &str) -> IntValue<I> {
     //     if I::sign() == Unsigned { LLVMBuildUDiv() } else { LLVMBuildSDiv() }
-    pub fn build_int_unsigned_div(&self, lhs: &IntValue, rhs: &IntValue, name: &str) -> IntValue {
+    pub fn build_int_unsigned_div<T: IntMathValue>(&self, lhs: &T, rhs: &T, name: &str) -> T {
```

Comment:
The references here are artifacts of a time when `*Value`s and `*Type`s weren't `Copy`. Since you're touching this code, I would greatly appreciate if you could change this and the other methods you modified to take ownership of lhs/rhs trait types.

IE, the signature should be `..., lhs: T, rhs: T, ...`

Thanks!

Reply:
Is it worth going through and changing all methods to take ownership, or would you rather I just changed the ones I touched?

Reply:
Just the ones you have touched - I have to look at the remainder on a case by case basis.